### ISSUE TITLE
Proposal: Separate sounds and lua code

### ISSUE THREAD

**Issue by Techbot121** [Metastruct/garrysmod-chatsounds#187]
As of now Chatsounds are using Capsadmins goluwa engine to process the Chatstrings.
I plan on moving out the parsing code from goluwa to a new separate instead, so we can make our own changes, rather than depending on goluwa (the gmod stable tag is about 300 commits behind master (!)).
This repo would then be then only used for the sounds themselves.

I created this issue so people can discuss 9and maybe provide some feedback.

Current feature wishlist:

- Proper compatibility with old Chatsounds
- Only fetch non existent or not mounted valve Chatsounds 
- ...

**Comment by PotcFdk** [comment]
If we ever actually give this a serious try, we should definitely add the sound set modularization plan into the operation. Given that this is a pretty big change, it makes sense to introduce all the maximum incompatibility changes all at once instead of spread out over multiple mid-sized annoyance moments.

The sub-goals of that plan would be
- modularizing the current repository
(I would be willing to work on this and/or to orchestrate the organization of stuff like maintainership)
- an in-game system to subscribe to and unsubscribe from an official list of sound repositories (the latter of which would be part of the orchestration from the previous item in this list)
- (eventually) providing documentation for third-parties on how to create their own curated chatsounds repositories and how to get them featured in the upstream lists

So basically, think of the next-generation chatsounds project to act like an app store, or rather, GNU/Linux-style package manager with configurable repositories.

**Comment by Earu** [comment]
@Techbot121 is this still valid with neo-cs?
